workspaces/ui-v2/src/pages/changelog/ChangelogListPage.tsx

```diff
@@ -41,16 +46,26 @@ export const ChangelogListPage: FC<
 };
 
 export function ChangelogRootPage(props: { changelogBatchId: string }) {
-  const { endpoints, loading } = useEndpoints(props.changelogBatchId);
+  const endpointsState = useAppSelector((state) => state.endpoints.results);
+  const changelog = useEndpointsChangelog(props.changelogBatchId);
+  const changelogByEndpointId = changelog.reduce(
+    (acc: Record<string, EndpointChangelog>, endpointChange) => {
+      acc[`${endpointChange.method}${endpointChange.pathId}`] = endpointChange;
+      return acc;
+    },
+    {}
+  );
```

Comment:
Nit: shall we have our own version of the "to reduce or not to reduce" debate that happened last week on Twitter 😂🙈? I don't really care either way, but try (and often forget)  to add a comment to my reduces to indicate what kind of operation it's implementing (a `group-by` in this case). Not that this is hard to deduce from reading this, but it seems to address many people's issues over reduce readability.

Reply:
Hmmm in this case, a group-by might be more intuitive, and I believe we are already using groupby - I'll update this part

Reply:
Ok I just tried implementing this, it's actually doing a `keyBy` instead of `groupBy` - in which we don't have the dependency installed (so I'm going to keep it as is)

In this case, I think the name is probably sufficient enough - but I agree, if there's anything more complicated with a reduce we should either:
- use a common library / util function
- annotate it with a comment

Reply:
It is a `indexBy` / `keyBy`, you're right! Kind of proving the point that a comment could instantly take away all guessing.  Apart from that, it's just a few lines of code without any surprising behaviours, so not sure that warrants extracting it / pulling in a util library.